### ISSUE TITLE
ibm_is_security_group_rule : removal of a tcp configuration results in no change

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#997]
Apply a ibm_is_security_group_rule with a tcp configuration like the one shown below.  Then delete the tcp configuration and apply again.  The tcp configuration is not removed.

```
resource "ibm_is_security_group_rule" "ingress_ssh_all" {
  group     = ibm_is_security_group.sg1.id
  direction = "inbound"
  remote    = "0.0.0.0/0" # TOO OPEN for production

/********************/
  tcp {
    port_min = 22
    port_max = 22
  }
/********************/
}

```
After apply the tfstate file still has the tcp configuration section and no changes were made to the ibm_is_security_group_rule:
```
{
      "mode": "managed",
      "type": "ibm_is_security_group_rule",
      "name": "ingress_ssh_all",
      "provider": "provider.ibm",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "direction": "inbound",
            "group": "r006-ff61287f-176a-4555-9dcf-0775ab567215",
            "icmp": [],
            "id": "r006-ff61287f-176a-4555-9dcf-0775ab567215.r006-9106612d-c231-418c-a9ed-6585a550ca2d",
            "ip_version": "ipv4",
            "remote": "0.0.0.0/0",
            "rule_id": "r006-9106612d-c231-418c-a9ed-6585a550ca2d",
            "tcp": [
              {
                "port_max": 22,
                "port_min": 22
              }
            ],
            "udp": []
          },
          "private": "bnVsbA==",
          "dependencies": [
            "ibm_is_security_group.sg1"
          ]
        }
      ]
    }
```

It is possible to work around this by tainting the ibm_is_security_group_rule:

```
➜  tfinstance git:(master) ✗ tf taint ibm_is_security_group_rule.ingress_ssh_all
Resource instance ibm_is_security_group_rule.ingress_ssh_all has been marked as tainted.
➜  tfinstance git:(master) ✗ tfa
data.ibm_is_image.ubuntu: Refreshing state...
...
```

After the taint, and then the apply the tcp configuration is gone:

```
{
      "mode": "managed",
      "type": "ibm_is_security_group_rule",
      "name": "ingress_ssh_all",
      "provider": "provider.ibm",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "direction": "inbound",
            "group": "r006-ff61287f-176a-4555-9dcf-0775ab567215",
            "icmp": [],
            "id": "r006-ff61287f-176a-4555-9dcf-0775ab567215.r006-cc0eb0a9-f8c0-4882-9e6e-06b03413c625",
            "ip_version": "ipv4",
            "remote": "0.0.0.0/0",
            "rule_id": "r006-cc0eb0a9-f8c0-4882-9e6e-06b03413c625",
            "tcp": [],
            "udp": []
          },
          "private": "bnVsbA==",
          "dependencies": [
            "ibm_is_security_group.sg1"
          ]
        }
      ]
    }
```
### Terraform Version
```
➜  tfinstance git:(master) ✗ tf version
Terraform v0.12.18
+ provider.ibm v1.0.0
```
### Terraform Configuration Files
First apply with this section:
```
/********************/
  tcp {
    port_min = 22
    port_max = 22
  }
/********************/
```

Then comment the tcp section out and apply again

```hcl
variable "ibmcloud_api_key" {
}

variable "ssh_key_name" {
}

variable "resource_group_name" {
}

variable "generation" {
  default = "2"
}

variable "ibmcloud_timeout" {
  description = "Timeout for API operations in seconds."
  default     = 900
}

variable "region" {
  default = "us-south"
}

variable "zone" {
  default = "us-south-1"
}

variable "basename" {
  description = "Name for the VPC to create and prefix to use for all other resources."
  default     = "example"
}

provider "ibm" {
  region           = var.region
  ibmcloud_api_key = var.ibmcloud_api_key
  ibmcloud_timeout = var.ibmcloud_timeout
  generation       = var.generation
}

data "ibm_resource_group" "group" {
  name = var.resource_group_name
}

resource "ibm_is_vpc" "vpc" {
  name           = var.basename
  resource_group = data.ibm_resource_group.group.id
}

resource "ibm_is_security_group" "sg1" {
  name           = "${var.basename}-sg1"
  vpc            = ibm_is_vpc.vpc.id
  resource_group = data.ibm_resource_group.group.id
}

# allow ssh access to this instance from anywhere on the planet
resource "ibm_is_security_group_rule" "ingress_ssh_all" {
  group     = ibm_is_security_group.sg1.id
  direction = "inbound"
  remote    = "0.0.0.0/0" # TOO OPEN for production

/********************/
  tcp {
    port_min = 22
    port_max = 22
  }
/********************/
}

resource "ibm_is_subnet" "subnet1" {
  name                     = "${var.basename}-subnet1"
  vpc                      = ibm_is_vpc.vpc.id
  zone                     = var.zone
  total_ipv4_address_count = 256
}

data "ibm_is_ssh_key" "ssh_key" {
  name = var.ssh_key_name
}

data "ibm_is_image" "ubuntu" {
  // name = "ubuntu-18.04-amd64"
  name = "ibm-ubuntu-18-04-1-minimal-amd64-1"
}

resource "ibm_is_instance" "vsi1" {
  name           = "${var.basename}-vsi1"
  vpc            = ibm_is_vpc.vpc.id
  zone           = var.zone
  keys           = [data.ibm_is_ssh_key.ssh_key.id]
  image          = data.ibm_is_image.ubuntu.id
  // profile        = "cc1-2x4"
  profile        = "cx2-2x4"
  resource_group = data.ibm_resource_group.group.id

  primary_network_interface {
    subnet          = ibm_is_subnet.subnet1.id
    security_groups = [ibm_is_security_group.sg1.id]
  }
}

resource "ibm_is_floating_ip" "fip1" {
  name   = "${var.basename}-fip1"
  target = ibm_is_instance.vsi1.primary_network_interface[0].id
}

output "sshcommand" {
  value = "ssh root@${ibm_is_floating_ip.fip1.address}"
}


terraform {
  required_version = ">= 0.12"
}

```

**Comment by hkantare** [comment]
This is working according the API ...Once you create any security rule you don't have option to edit the protocol
<img width="452" alt="Screen Shot 2020-01-09 at 12 27 49 PM" src="https://user-images.githubusercontent.com/24930306/72045250-aa067d00-32db-11ea-8d55-5c42221628f2.png">


ic is security-group-rule-update GROUP RULE_ID [--direction DIRECTION] [--remote REMOTE_ADDRESS | CIDR_BLOCK | SECURITY_GROUP_ID] [--icmp-type ICMP_TYPE [--icmp-code ICMP_CODE]] [--port-max PORT_MAX --port-min PORT_MIN] [--json]

**Comment by powellquiring** [comment]
can this be treated as a change in the configuration that requires the resource to be delete/create?

**Comment by hkantare** [comment]
Fixed in latest releases v0.23.0 and v1.1.0
https://github.com/IBM-Cloud/terraform-provider-ibm/releases
